src/components/FileMenu/RenameDialog.js

```diff
@@ -0,0 +1,97 @@
+import React, { useMemo, useState } from 'react'
+
+import PropTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { useDataMutation } from '@dhis2/app-runtime'
+import {
+    Modal,
+    ModalTitle,
+    ModalContent,
+    ModalActions,
+    ButtonStrip,
+    Button,
+    InputField,
+    TextAreaField,
+} from '@dhis2/ui'
+
+import { supportedFileTypes, endpointFromFileType } from './utils'
+
+const getMutation = type => ({
+    resource: endpointFromFileType(type),
+    id: ({ id }) => id,
+    type: 'update',
+    partial: true,
+    data: ({ name, description }) => ({ name, description }),
+})
+
+export const RenameDialog = ({ type, object, onClose, onRename, onError }) => {
+    const [name, setName] = useState(object.name)
+    const [description, setDescription] = useState(object.description)
+
+    const mutation = useMemo(() => getMutation(type), [])
```

Comment:
Curious if there was a performance issue that triggered memoizing the mutation?

Reply:
The problem was passing the `type` and getting back a different function, otherwise a mutation definition for each type should have been used which is not feasible.
This is the workaround suggested by @amcgee.

Reply:
@edoardo I believe eslint will complain here unless `[type]` is in the dependencies?  We do want to update the mutation if the type is ever changed, however it won't be picked up by the mutation hook - so what you have here is technically correct, just curious why eslint isn't complaining.